packages/builders/src/messages/Message.ts

```diff
@@ -0,0 +1,544 @@
+	 * ```ts
+	 * message.spliceAttachments(0, 1);
+	 * ```
+	 * @example
+	 * Remove the first n attachments:
+	 * ```ts
+	 * const n = 4;
+	 * message.spliceAttachments(0, n);
+	 * ```
+	 * @example
+	 * Remove the last attachment:
+	 * ```ts
+	 * message.spliceAttachments(-1, 1);
+	 * ```
+	 * @param start - The index to start at
+	 * @param deleteCount - The amount of attachments to remove
+	 * @param attachments - The attachments to insert
+	 */
+	public spliceAttachments(
+		start: number,
+		deleteCount: number,
+		...attachments: RestOrArray<APIAttachment | AttachmentBuilder | ((builder: AttachmentBuilder) => AttachmentBuilder)>
+	): this {
+		const resolved = normalizeArray(attachments).map((attachment) => resolveBuilder(attachment, AttachmentBuilder));
+		this.data.attachments.splice(start, deleteCount, ...resolved);
+
+		return this;
+	}
+
+	/**
+	 * Sets the flags for this message.
+	 */
+	public setFlags(flags: number): this {
+		this.data.flags = flags;
+		return this;
+	}
+
+	/**
+	 * Clears the flags for this message.
+	 */
+	public clearFlags(): this {
+		this.data.flags = undefined;
+		return this;
+	}
+
+	/**
+	 * Sets `enforce_nonce` for this message.
+	 */
+	public setEnforceNonce(enforceNonce = true): this {
+		this.data.enforce_nonce = enforceNonce;
+		return this;
+	}
+
+	/**
+	 * Sets the poll for this message.
+	 *
+	 * @param poll - The poll to set
+	 */
+	public setPoll(poll: APIPoll | PollBuilder | ((builder: PollBuilder) => PollBuilder)): this {
+		this.data.poll = resolveBuilder(poll, PollBuilder);
+		return this;
+	}
+
+	/**
+	 * Updates the poll for this message (and creates it if it doesn't exist)
+	 *
+	 * @param updater - The function to update the poll with
+	 */
+	public updatePoll(updater: (builder: PollBuilder) => PollBuilder): this {
+		this.data.poll = updater(this.data.poll ?? new PollBuilder());
+		return this;
+	}
+
+	/**
+	 * Clears the poll for this message.
+	 */
+	public clearPoll(): this {
+		this.data.poll = undefined;
+		return this;
+	}
+
+	/**
+	 * Serializes this builder to API-compatible JSON data.
+	 *
+	 * Note that by disabling validation, there is no guarantee that the resulting object will be valid.
+	 *
+	 * @param validationOverride - Force validation to run/not run regardless of your global preference
+	 */
+	public toJSON(validationOverride?: boolean): RESTPostAPIChannelMessageJSONBody {
+		const { poll, allowed_mentions, attachments, embeds, components, message_reference, ...rest } = this.data;
+
+		const data = {
+			...structuredClone(rest),
+			// Covered by the messagePredicate
+			poll: this.data.poll?.toJSON(false),
+			allowed_mentions: allowed_mentions?.toJSON(false),
+			attachments: attachments.map((attachment) => attachment.toJSON(false)),
+			embeds: this.data.embeds.map((embed) => embed.toJSON(false)),
+			components: this.data.components?.map((component) => component.toJSON(validationOverride)),
+			message_reference: message_reference?.toJSON(false),
```

Comment:
So why are some false and some validationOverride? 👀

Reply:
On purpose. The messagePredicate does specific assertions that the particular component types are used, as opposed to just using the componentPredicate.

Will update the comment regardless.